### PR TITLE
Add lsp support for ruby

### PR DESCRIPTION
**Pull request by przbadu**
```sh
:LspInstall ruby
```

### CODE REVIEW

File: lua/lsp/ruby-ls.lua [hunk 1]
```diff
@@ -0,0 +1,5 @@
+-- If you are using rvm, make sure to change below configuration
+require'lspconfig'.solargraph.setup {
+    cmd = { DATA_PATH .. "~/.rbenv/shims/solargraph", "--stdio" },
+    on_attach = require'lsp'.common_on_attach
+}
```
Comment (ChristianChiarulli) [inline lua/lsp/ruby-ls.lua:5]: Consider adding the handler, you can find it in the python lsp module and then pass the ruby diagnostic values here

Comment (ChristianChiarulli) [inline lua/lsp/ruby-ls.lua:5]: consider adding the filetypes field as well.

File: lua/lsp/ruby-ls.lua [hunk 2]
```diff
@@ -0,0 +1,5 @@
+-- If you are using rvm, make sure to change below configuration
+require'lspconfig'.solargraph.setup {
+    cmd = { DATA_PATH .. "~/.rbenv/shims/solargraph", "--stdio" },
```
Comment (ChristianChiarulli) [inline lua/lsp/ruby-ls.lua:3]: you could also make the solargraph path configurable and expose that in the settings for the user if they would like it to point to a virtuual env or binary found in the PATH

**Review by ChristianChiarulli — CHANGES_REQUESTED** [review]
This is a good start, consider adding the handler as well for

File: lua/lsp/ruby-ls.lua (hunk 2 above)
Comment (przbadu) [reply lua/lsp/ruby-ls.lua:3]: Hey, thank you for reviewing my PR, I am totally new to nvim and lua, I might need to look into lua soon. :smile: 

can you please suggest me on how to make solargraph path configurable? I am trying to Fix these reviews, I really love what you have done with LunarVim. It is easily configurable for my need.

Comment (ChristianChiarulli) [reply lua/lsp/ruby-ls.lua:3]: sure, I think what we have now is probably good enough, but you could just have a setting like:

 `O.ruby.solargraph_path="~/.rbenv/shims/solargraph"`

Then let the user override this in lv-settings

**Review by ChristianChiarulli — APPROVED** [review]

**Pull request merged**